ams/server/src/main/java/com/netease/arctic/server/optimizing/UnKeyedTableCommit.java

```diff
@@ -186,14 +189,18 @@ protected void replaceFiles(
       if (CollectionUtils.isNotEmpty(removedDataFiles) ||
           CollectionUtils.isNotEmpty(addedDataFiles)) {
         RewriteFiles dataFileRewrite = transaction.newRewrite();
+        // original snapshot might be expired
+        Snapshot targetSnapshot = table.asUnkeyedTable().snapshot(targetSnapshotId) == null ?
+            table.asUnkeyedTable().currentSnapshot() :
+            table.asUnkeyedTable().snapshot(targetSnapshotId);
```

Comment:
The `targetSnapshot` should not be the current snapshot, which cannot achieve the goal of checking the commit conflicts.

Maybe we should open another PR to fix the commit failure issue caused by the expiration of the target snapshot.

Reply:
make sense